builder/amazon/common/access_config.go

```diff
@@ -18,53 +15,69 @@ import (
 // AccessConfig is for common configuration related to AWS access
 type AccessConfig struct {
 	AccessKey         string `mapstructure:"access_key"`
-	SecretKey         string `mapstructure:"secret_key"`
+	CustomEndpointEc2 string `mapstructure:"custom_endpoint_ec2"`
+	MFACode           string `mapstructure:"mfa_code"`
+	ProfileName       string `mapstructure:"profile"`
 	RawRegion         string `mapstructure:"region"`
+	SecretKey         string `mapstructure:"secret_key"`
 	SkipValidation    bool   `mapstructure:"skip_region_validation"`
 	Token             string `mapstructure:"token"`
-	ProfileName       string `mapstructure:"profile"`
-	CustomEndpointEc2 string `mapstructure:"custom_endpoint_ec2"`
+	session           *session.Session
 }
 
 // Config returns a valid aws.Config object for access to AWS services, or
 // an error if the authentication and region couldn't be resolved
-func (c *AccessConfig) Config() (*aws.Config, error) {
-	var creds *credentials.Credentials
+func (c *AccessConfig) Session() (*session.Session, error) {
+	if c.session != nil {
+		return c.session, nil
+	}
 
 	region, err := c.Region()
 	if err != nil {
 		return nil, err
 	}
-	config := aws.NewConfig().WithRegion(region).WithMaxRetries(11)
+
+	if c.ProfileName != "" {
```

Comment:
I'm not seeing this anywhere in Rickard's original PR or in master.  Where is it coming from?